extensions/amp-pan-zoom/0.1/test/test-amp-pan-zoom.js

```diff
@@ -0,0 +1,265 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../amp-pan-zoom';
+
+describes.realWin('amp-pan-zoom', {
+  amp: {
+    extensions: ['amp-pan-zoom'],
+  },
+}, env => {
+  let win, doc, sandbox;
+  let el, impl;
+  let svg;
+
+  const measureMutateElementStub = (measure, mutate) => {
+    let result = Promise.resolve();
+    if (measure) {
+      result = result.then(measure);
+    }
+    if (mutate) {
+      result = result.then(mutate);
+    }
+    return result;
+  };
+
+  /**
+   * This function takes an object of attributes and constructs and builds
+   * an amp-pan-zoom element and attaches it to the page. Assumes the contents
+   * to be an SVG of width and height 300x300.
+   * @param {Object} opt_attributes
+   */
+  function getPanZoom(opt_attributes) {
+    el = doc.createElement('amp-pan-zoom');
+    // Default attributes that can be overidden
+    el.setAttribute('layout', 'fixed');
+    el.setAttribute('width', '300');
+    el.setAttribute('height', '400');
+
+    for (const key in opt_attributes) {
+      el.setAttribute(key, opt_attributes[key]);
+    }
+
+    svg = doc.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    svg.setAttribute('width', '100');
+    svg.setAttribute('height', '100');
+    const svgNS = svg.namespaceURI;
+    const rect = doc.createElementNS(svgNS,'rect');
+    rect.setAttribute('x', '0');
+    rect.setAttribute('y', '0');
+    rect.setAttribute('width','100');
+    rect.setAttribute('height','100');
+    rect.setAttribute('fill','#95B3D7');
+    svg.appendChild(rect);
+
+    el.appendChild(svg);
+    doc.body.appendChild(el);
+    return el.build().then(() => {
+      impl = el.implementation_;
+      sandbox.stub(impl, 'measureMutateElement')
+          .callsFake(measureMutateElementStub);
+      sandbox.stub(impl, 'mutateElement')
+          .callsFake(mutate => {
+            mutate();
+            return Promise.resolve();
+          });
+    });
+  }
+
+  beforeEach(() => {
+    win = env.win;
+    doc = win.document;
+    sandbox = env.sandbox;
+    env.iframe.height = 500;
+    env.iframe.width = 400;
+  });
+
+  it('should initialize all attributes correctly', () => {
+    return getPanZoom({
+      'max-scale': '4',
+      'initial-scale': '2',
+      'initial-x': '1',
+      'initial-y': '5',
+      'reset-on-resize': '',
+      'disable-double-tap': '',
+    }).then(() => {
+      expect(impl.initialScale_).to.equal(2);
+      expect(impl.initialX_).to.equal(1);
+      expect(impl.initialY_).to.equal(5);
+      expect(impl.maxScale_).to.equal(4);
+      expect(impl.resetOnResize_).to.equal(true);
+      expect(impl.disableDoubleTap_).to.equal(true);
+    });
+  });
+
+  it('should size contents correctly', () => {
+    return getPanZoom().then(() => el.layoutCallback()).then(() => {
+      expect(svg.style.width).to.equal('300px');
```

Comment:
Would be better to check the actual rect. For example, if you somehow had some CSS that set width to `400px !important`, your test would still pass.

Reply:
Changed this to use `clientWidth` and `clientHeight` instead.